api/external.go

```diff
@@ -260,6 +261,14 @@ func (a *API) internalExternalProviderCallback(w http.ResponseWriter, r *http.Re
 				}); terr != nil {
 					return terr
 				}
+				logger.LogEntrySetFields(r, logrus.Fields{
```

Comment:
not too sure how we can reduce duplicated code here... ideally we'll want to include the auth event in the logs everytime there is a new row insert into the `audit_log_entries` table

another solution i thought of was to pass the `http.Request` as an argument into `NewAuditLogEntry` (https://github.com/supabase/gotrue/pull/562/files#diff-57bd73e87d7b244be8c43b1e9acf055f2bf61b9f714cffa760298b3314bce6e6R66) so we can just set the payload in the log fields using `logger.LogEntrySetFields(request, payload) - would this be a cleaner approach to reduce code duplication?

Reply:
cc @hf @J0

Reply:
Not sure if it's a permissions thing but can't quite see the Github diff: https://github.com/supabase/gotrue/pull/562/files#diff-57bd73e87d7b244be8c43b1e9acf055f2bf61b9f714cffa760298b3314bce6e6R66 

Am OK with either solution  --   passing in `http.Request` to `NewAuditLogEntry` or having code duplication. If we  really wanted to try and cut back on code I suppose we could use a function to generate the auth event struct

```go
func generateEvent(actorID, actorUsername .....) {
                         return logrus.Fields{
 						"actor_id":       actorID,
 						"actor_username": user.GetEmail(),
 						"action":         models.UserSignedUpAction,
 						"log_type":       models.ActionLogTypeMap[models.UserSignedUpAction],
 					}

}
```

will defer to @hf  here

Reply:
went with passing the request into `NewAuditLogEntry` so we don't have to add `logger.LogEntrySetFields` everywhere